src/app.ts

```diff
@@ -122,4 +184,135 @@ app.get(
+      await verifyFCMToken(req.query.playerToken).catch((err) => {
+        console.error(err)
+        correctToken = false
+      })
+    }
+    if (!correctToken) {
+      return res.sendStatus(400)
+    }
+    const client = getClient()
+
+    client
+      .connect()
+      .then(async () => {
+        const checkIfGameExistsQuery = // todo add check if game is full
+          'SELECT game_master FROM Games WHERE game_id=$1'
+        const gameCheckValues = [req.query.gameId]
+        const createPlayerQuery =
+          'INSERT INTO Players(token, nickname, turn, game_id, card1, card2, funds, bet) VALUES($1, $2, $3, $4, $5, $6, $7, $8)'
+        const createPlayerValues = [
+          req.query.playerToken,
+          req.query.nickname,
+          0,
+          req.query.gameId,
+          null,
+          null,
+          null,
+          null,
+        ]
+        const getGameInfoQuery = 'SELECT * FROM Games WHERE game_id=$1'
+        const getGameInfoValues = [req.query.gameId]
+        const getPlayersInRoomQuery =
+          'SELECT nickname, token FROM Players WHERE game_id=$1'
+        const getPlayersInRoomValues = [req.query.gameId]
+        await client
+          .query(checkIfGameExistsQuery, gameCheckValues)
+          .then(async (result) => {
+            if (result.rowCount === 0) {
+              // game does not exist
+              return res.sendStatus(401)
+            } else {
+              // game exists
+              const gameInfo: gameSettings = {
+                smallBlind: 0,
+                startingFunds: 0,
+                players: [],
+                gameMasterHash: sha256(result.rows[0].game_master).toString(),
+              }
+              await client
+                .query(createPlayerQuery, createPlayerValues)
+                .catch((err) => {
+                  console.error(err.stack)
+                  return res.sendStatus(500)
+                })
+              await client
+                .query(getGameInfoQuery, getGameInfoValues)
+                .then((result) => {
+                  gameInfo.smallBlind = parseInt(result.rows[0].small_blind)
+                  gameInfo.startingFunds = parseInt(
+                    result.rows[0].starting_funds
+                  )
+                })
+                .catch((err) => {
+                  console.error(err.stack)
+                  return res.sendStatus(500)
+                })
+              await client
+                .query(getPlayersInRoomQuery, getPlayersInRoomValues)
+                .then((result) => {
+                  const message = {
+                    data: {
+                      type: 'playerJoined',
+                      nickname: req.query.nickname as string,
+                      playerHash: sha256(req.query.playerToken).toString(),
+                    },
+                    token: '',
+                  }
+                  // We know that the values will be defined
```

Comment:
I don't get this comment, what values will be defined? The `result.rows` will be defined because we are in happy branch of db request

Reply:
What I meant was that nickname is defined, because req.query.nickname can potentially be undefined, but we checked it with celebrate. That is why we can use 'as string'.